handlers/delete_lease_test.go

```diff
@@ -144,3 +144,55 @@ func TestDeleteLeaseExists(t *testing.T) {
 		}
 	}
 }
+
+func TestDeleteLeaseExistsNoClearNamespaces(t *testing.T) {
+	clusterNames := []string{"cluster1", "cluster2"}
+	uuids := make([]uuid.UUID, len(clusterNames))
+	annos := testutil.GetRawAnnotations(
+		clusterNames,
+		leases.TimeFormat,
+		func(i int) time.Time {
+			return time.Now().Add(1 * time.Hour)
+		},
+		func(i int) uuid.UUID {
+			ret := uuid.NewUUID()
+			uuids[i] = ret
+			return ret
+		},
+	)
+	for i, u := range uuids {
+		getterUpdater := newFakeServiceGetterUpdater(
+			&api.Service{ObjectMeta: api.ObjectMeta{Annotations: annos}},
+			nil,
+			nil,
+			nil,
+		)
+		path := "/lease/" + u.String()
+
+		defaultNamespaces := []string{"default", "kube-system"}
+		namespaces := append(defaultNamespaces, "ns1", "ns2")
+		nsList := api.NamespaceList{}
+		for _, namespace := range namespaces {
+			nsList.Items = append(nsList.Items, api.Namespace{ObjectMeta: api.ObjectMeta{Name: namespace}})
+		}
+
+		listClusterResp := newListClusterResp(clusterNames)
+		clusterLister := newFakeClusterLister(listClusterResp, nil)
+		nsListerDeleter := newFakeNamespaceListerDeleter(&nsList, nil, nil)
+		hdl := DeleteLease(getterUpdater, clusterLister, "claimer", "proj1", "zone1", false, getNSFunc(nsListerDeleter, nil))
+		req, err := http.NewRequest("DELETE", path, nil)
+		req.Header.Set("Authorization", "some awesome token")
+		if err != nil {
+			t.Errorf("trial %d: error creating request %s (%s)", i, path, err)
+			continue
+		}
+		res := httptest.NewRecorder()
+		hdl.ServeHTTP(res, req)
+		if res.Code != http.StatusOK {
+			t.Errorf("trial %d: status code for path %s was %d, not %d", i, path, res.Code, http.StatusOK)
+			continue
+		}
+		// ensure namespaces were not deleted
+		assert.Equal(t, len(nsListerDeleter.NsDeleted), 0, "number of deleted namespaces")
+	}
+}
```

Comment:
this is the same test as `TestDeleteLeaseExists` above, except it specifies that the `DeleteLease` handler should _not_ clear namespaces